### PR TITLE
Enhance cmdline handling in linux-qcom-uki.bb

### PR DESCRIPTION
**Pull request by quic-vkraleti**
Construct kernel cmdline by reading QCOM_BOOTIMG_ROOTFS, SERIAL_CONSOLES and KERNEL_CMDLINE_EXTRA variables and update the same into uki cmdline.

### CODE REVIEW

File: recipes-kernel/images/linux-qcom-uki.bb [hunk 1]
```diff
@@ -51,10 +51,28 @@ do_compile() {
     ukify_cmd="$ukify_cmd --uname ${KERNEL_VERSION}"
 
     # Kernel cmdline
+    cmdline=""
+    if [ -n "${QCOM_BOOTIMG_ROOTFS}" ]; then
+        cmdline="$cmdline root=${QCOM_BOOTIMG_ROOTFS} rw rootwait"
+    fi
```
Comment (lumag) [inline recipes-kernel/images/linux-qcom-uki.bb:57]: `QCOM_BOOTIMG_ROOTFS` varies per dtb. Please take that into account.

File: recipes-kernel/images/linux-qcom-uki.bb [hunk 2]
```diff
@@ -51,10 +51,28 @@ do_compile() {
...
+
+    if [ ! -z "${SERIAL_CONSOLES}" ]; then
+        tmp="${SERIAL_CONSOLES}"
+        for entry in $tmp ; do
+            baudrate=`echo $entry | sed 's/\;.*//'`
+            tty=`echo $entry | sed -e 's/^[0-9]*\;//' -e 's/\;.*//'`
+	    console="$tty","$baudrate"n8
+            cmdline="$cmdline console=$console"
+        done
+    fi
```
Comment (lumag) [inline recipes-kernel/images/linux-qcom-uki.bb:67]: We have similar code in `linux-qcom-bootimg.bbclass`, please consider extracting a function to OE-Core / `kernel.bbclass`.

For the reference:
```Python
        consoles = ' '.join(map(lambda c: "console=%(tty)s,%(rate)sn8" % dict(zip(("rate", "tty"), c.split(';'))), getVarDTB("SERIAL_CONSOLES").split()))
```

**Review by lumag — CHANGES_REQUESTED** [review]

File: recipes-kernel/images/linux-qcom-uki.bb (hunk 2 above)
Comment (quic-vkraleti) [reply recipes-kernel/images/linux-qcom-uki.bb:67]: KERNEL_CMDLINE_EXTRA is a meta-qcom variable. OE-Core kernel.bbclass don't support updating cmdline. For now do consider kernel cmdline setting to be a meta-qcom specific implementation only. I'll try to submit a proposal to oe-core but not sure how far it'll fly.

File: recipes-kernel/images/linux-qcom-uki.bb (hunk 1 above)
Comment (quic-vkraleti) [reply recipes-kernel/images/linux-qcom-uki.bb:57]: @lumag if you recollect discussions in https://github.com/Linaro/meta-qcom/pull/548, we agreed to make UKI, dtb agnostic and keep a separate dtb section in ESP. We also discussed an approach to have a new partition to host dtb files.  So updating cmdline in UKI that varies per dtb won't be feasible.

Comment (quic-vkraleti) [reply recipes-kernel/images/linux-qcom-uki.bb:57]: @lumag submitted PR: https://github.com/Linaro/meta-qcom/pull/620 to move DTB away from esp and create a separate dtb image. With this dtb and uki are completely isolated.

File: recipes-kernel/images/linux-qcom-uki.bb (hunk 2 above)
Comment (lumag) [reply recipes-kernel/images/linux-qcom-uki.bb:67]: Ack. Please cc me if you post such a patch.

File: recipes-kernel/images/linux-qcom-uki.bb (hunk 1 above)
Comment (lumag) [reply recipes-kernel/images/linux-qcom-uki.bb:57]: The issue is that all those variables wary from machine to machine. So you end up with an image that is expected to be generic, but in reality it isn't.

Comment (lumag) [reply recipes-kernel/images/linux-qcom-uki.bb:57]: I don't have a good solution for the issue. maybe a setup-specific DT overlay to be applied by the EFI/bootloader. Maybe a bootloader should set them.

Comment (lumag) [reply recipes-kernel/images/linux-qcom-uki.bb:57]: I'm going to merge this, but let's think for a longer-term plan.

**Pull request merged**